home-task-2_test.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"math"
+	"testing"
+)
+
+func Test_FigureOne(t *testing.T) {
+	var s Figure = Square{10}
+	var c Figure = Circle{10}
+
+	if s.area() != 10 * 10 {
+		t.Error("Square area incorrect")
+	}
+	if s.perimeter() != 10 * 4 {
+		t.Error("Square perimeter incorrect")
+	}
+
+	if c.area() != math.Pi * 10 * 10 {
+		t.Error("Circle area incorrect")
+	}
+
+	if c.perimeter() != 2 * math.Pi * 10 {
+		t.Error("Circle perimeter incorrect")
+	}
+}
+
+func Test_FigureTwo(t *testing.T) {
+	var s Figure = Square{}
+	var c Figure = Circle{}
+
+	if s.area() != 0 {
```

Comment:
this is a bit confusing, could be clered up, but that's not important atm.

Reply:
29 and 30 lines taken from PDF lesson

32 check that returned area is not equal to 0, function has errors, since initial value of calculation is 0, area is also 0 😏